conans/client/tools/win.py

```diff
@@ -68,26 +69,94 @@ def vs_installation_path(version):
 
     if version not in vs_installation_path._cached:
         vs_path = None
-        program_files = os.environ.get("ProgramFiles(x86)", os.environ.get("ProgramFiles"))
-        if program_files:
-            vswhere_path = os.path.join(program_files, "Microsoft Visual Studio", "Installer",
-                                        "vswhere.exe")
-            if os.path.isfile(vswhere_path):
-                version_range = "[%d.0, %d.0)" % (int(version), int(version) + 1)
-                try:
-                    output = subprocess.check_output([vswhere_path, "-version", version_range,
-                                                      "-legacy", "-property", "installationPath"])
-                    vs_path = output.decode().strip()
-                    _global_output.info("vswhere detected VS %s in %s" % (version, vs_path))
-                except (ValueError, subprocess.CalledProcessError, UnicodeDecodeError) as e:
-                    _global_output.error("vswhere error: %s" % str(e))
+        output_legacy = json.loads(vswhere(legacy=True, format_="json"))
+        output_products = json.loads(vswhere(products=["*"], format_="json"))
+        output = output_legacy + output_products
+
+        for installation in output:
+            if installation["installationVersion"].startswith(("%d." % int(version))):
+                vs_path = installation["installationPath"]
 
         # Remember to cache result
         vs_installation_path._cached[version] = vs_path
 
     return vs_installation_path._cached[version]
 
 
+def vswhere(all_=False, prerelease=False, products=None, requires=None, version="", latest=False,
+            legacy=False, format_="", property_="", nologo=True):
+
+    # 'version' option only works if Visual Studio 2017 is installed:
```

Comment:
What is the output of this function? list of paths?

Reply:
I could be a list of `products` (json objects with lots of fileds: `installationPath`, `productId`...), or if there is just one product it returns just that json object.

Reply:
Then I think we should unify the return and return always a list, even with one element, check the return type is a nightmare